linera-execution/src/wasm/wasmer.rs

```diff
@@ -100,22 +100,50 @@ static WASMER_INSTANTIATE_LOCK: Lazy<std::sync::Mutex<()>> =
     Lazy::new(|| std::sync::Mutex::new(()));
 
 impl WasmApplication {
-    /// Prepares a runtime instance to call into the WASM contract.
-    pub fn prepare_contract_runtime_with_wasmer<'runtime>(
-        &self,
-        runtime: &'runtime dyn ContractRuntime,
-    ) -> Result<WasmRuntimeContext<'static, Contract<'runtime>>, WasmExecutionError> {
-        let metering = Arc::new(Metering::new(
-            runtime.remaining_fuel(),
-            Self::operation_cost,
-        ));
+    /// Creates a new [`WasmApplication`] using Wasmtime with the provided bytecodes.
+    pub fn new_with_wasmer(
+        contract_bytecode: Bytecode,
+        service_bytecode: Bytecode,
+    ) -> Result<Self, WasmExecutionError> {
+        let contract_engine = Self::create_wasmer_engine_for_contracts(0);
+        let contract_module = Module::new(&contract_engine, contract_bytecode)
+            .map_err(wit_bindgen_host_wasmer_rust::anyhow::Error::from)?;
+
+        let service_engine = Self::create_wasmer_engine_for_services();
+        let service_module = Module::new(&service_engine, service_bytecode)
+            .map_err(wit_bindgen_host_wasmer_rust::anyhow::Error::from)?;
+
+        Ok(WasmApplication::Wasmer {
+            contract_engine,
+            contract_module,
+            service_engine,
+            service_module,
+        })
+    }
+
+    /// Creates an [`Engine`] instance configured to run application contracts.
+    fn create_wasmer_engine_for_contracts(fuel: u64) -> Engine {
+        let metering = Arc::new(Metering::new(fuel, Self::operation_cost));
```

Comment:
This argument `fuel` is surprising.

Reply:
Good point, I had added this only to reuse the method in `prepare_contract_runtime_with_wasmer`, but that code was removed because the `Engine` couldn't be discarded. Since it's no longer necessary, I removed it :+1: